src/pumipic_adjacency.tpp

```diff
@@ -395,16 +395,25 @@ namespace pumipic {
     printf("Min area is: %.15f, Planned tol is %.15f\n", min_area, tol);
     return tol;
   }
-  
-  template <class ParticleType, typename Segment3d, typename SegmentInt>
-  bool search_mesh(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
+
+  template <class ParticleType, typename Segment3d, typename SegmentInt, typename Func>
+  bool particle_search(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
```

Comment:
@cwsmith @Angelyr any suggestions on the naming of this search method? Maybe something more descriptive like, `adjacency_based_particle_update`?

Reply:
A few ideas... none are great:
`find_ptcl_bounding_element`
`update_ptcl_bounding_element`
`update_element_association_of_particles` 
`set_particle_to_element_association`

Reply:
I think this is the last thing to resolve before we can merge. How about `trace_particle_through_mesh` ?

Reply:
That works.  We should put some doxygen style comments on the function and explicitly describe what it does.